engine/orchestration/src/com/cloud/vm/VirtualMachinePowerStateSyncImpl.java

```diff
@@ -111,7 +111,19 @@ private void processReport(long hostId, Map<Long, VirtualMachine.PowerState> tra
 
             for (VMInstanceVO instance : vmsThatAreMissingReport) {
 
-                Date vmStateUpdateTime = instance.getUpdateTime();
+                // Make sure powerState is up to date for missing VMs
+                try {
+                    if (!_instanceDao.isPowerStateUpToDate(instance.getId())) {
+                        s_logger.warn("Detected missing VM but power state is outdated, wait for another process report run for VM id: " + instance.getId());
+                        _instanceDao.resetVmPowerStateTracking(instance.getId());
+                        continue;
+                    }
+                } catch (CloudRuntimeException e) {
+                    s_logger.warn("Checked for missing powerstate for none existing vm", e);
```

Comment:
Can you correct the sentence to "Checked for missing powerstate of a non existing vm"?